Format using `group_imports`

grtlr opened the pull request:
# Description of change

This reformats everything using `group_format = "StdExternalCrate"`.

Closes #1234.

## Change checklist

<!-- Tick the boxes that are relevant to your changes, and delete any items that are not. -->

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

thibault-martinez commented:
Let's merge that after #1234 to avoid changing too much files many times.

grtlr committed:
Format using `group_imports`

grtlr committed:
Redo `edition` and sort alphabetically

Alex6323 commented on rustfmt.toml:
```diff
@@ -1,9 +1,11 @@
 version = "Two"
 comment_width = 120
+edition = "2021"
```
Is this necessary?

thibault-martinez replied:
It's not mandatory for most cases, only when not going through cargo fmt itself and I believe that's the case for on-save fmt on most IDEs. Doesn't hurt to have it anyway.

grtlr committed:
Automatically add `self` using regex

The following regex was used:
```
(mod ([a-z_]+)\b;((.|\n)*)?(use|pub\suse)\s)\2::
```
We can then use `$1self::$2::` to replace the `self`.

grtlr committed:
Format after replacing

grtlr committed:
Use improved regex

```
(mod ([a-z_]+)\b;((.|\n)*)(use|pub\suse)\s)\2::
```

grtlr committed:
Reformat

thibault-martinez committed:
Merge branch 'mainnet-develop' into mainnet-group-fmt

thibault-martinez committed:
Put self:: everywhere
